Expand on stale handling for gatsby sites (#22230)
gatsbyjs/gatsby: docs/docs/gatsby-link.md

}
```

## Handling stale client-side pages

> Gatsby's `<Link>` component will only fetch each page's resources once, when it doesn't have a cached copy. Updates to pages on the site are not reflected in the browser as they are effectively "locked in time". This can have the undesirable impact of different users having different views of the content.

Reviewer: ```suggestion
Gatsby's `<Link>` component will only fetch each page's resources once. Updates to pages on the site are not reflected in the browser as they are effectively "locked in time". This can have the undesirable impact of different users having different views of the content.
```
Wondering if it'd make more sense to add a note about not fetching anything if there is a cached copy?

Author: I think it's actually covered by paragraph opening "This only works if the page..."

> In order to prevent this staleness, Gatsby also requests an additional resource on each new page load: `app-data.json`. This contains a hash generated when the site is built; if anything in the `src` directory changes, the hash will change. During page loads, if Gatsby sees a different hash in the `app-data.json` to the hash it initially retrieved when the site first loaded then the browser will navigate using `window.location`. The browser fetches the new page and starts over again, so any cached resources are lost.

Reviewer: ```suggestion
In order to prevent this staleness, Gatsby requests an additional resource on each new page load: `app-data.json`. This contains a hash generated when the site is built; if anything in the `src` directory changes, the hash will change. During page loads, if Gatsby sees a different hash in the `app-data.json` than the hash it initially retrieved when the site first loaded, the browser will navigate using `window.location`. The browser fetches the new page and starts over again, so any cached resources are lost.
```

> This only works if the page has not previously been loaded, as the `app-data.json` will not be re-requested for that page. Therefore staleness isn't a solved problem, but it should be a less frequent one.

Author: > ... loaded content will be fetched.

I think "previously loaded content will be used" is more accurate. Gatsby won't refetch what is has cached.

> This only works if the page has not previously been loaded, as the `app-data.json` will not be re-requested for that page. Therefore staleness isn't a solved problem, but it should be a less frequent one.

Reviewer: Makes sense. Let's go with that then.

In order to prevent this staleness, Gatsby requests an additional resource on each new page load: `app-data.json`. This contains a hash generated when the site is built; if anything in the `src` directory changes, the hash will change. During page loads, if Gatsby sees a different hash in the `app-data.json` than the hash it initially retrieved when the site first loaded, the browser will navigate using `window.location`. The browser fetches the new page and starts over again, so any cached resources are lost.

> However, if the page has previously loaded, it will not re-request `app-data.json`. In that case, the hash comparison will not occur and the previously loaded content will be used. Therefore staleness isn't a solved problem, but it should be a less frequent one.

Reviewer: ```suggestion
However, if the page has previously loaded, it will not re-request `app-data.json`. In that case, the hash comparison will not occur and the previously loaded content will be used. 
```
I believe this is an accurate statement, but it's a bit more editorial than usually included in the docs. It seems as if explaining the potentially challenging case is sufficient?
